src/lxc/lxc_attach.c

```diff
@@ -273,7 +273,12 @@ static int get_pty_on_host(struct lxc_container *c, struct wrapargs *wrap, int *
 	if (lxc_console_create(conf) < 0)
 		return -1;
 	ts = conf->console.tty_state;
-	ts->stdoutfd = STDOUT_FILENO;
+	/* If stdout is not a tty but stderr is, we write to stderr. */
+	if (!isatty(STDOUT_FILENO) && isatty(STDERR_FILENO))
+		ts->stdoutfd = STDERR_FILENO;
+	else
+		ts->stdoutfd = STDOUT_FILENO;
+
```

Comment:
> we need to make sure to let any valid output from container reach the console, so we send it to fd 2.

That's the bit that I don't understand. If a command (run in the container) is only outputting stuff to its stdout, and lxc-attach's stdout is redirected into a file, that command stdout output should go into that file, and nowhere else. Why should it suddenly go to the host's (i. e. lxc-attach's) stderr? Similarly, the container command's stderr data should go (only) to lxc-attach's stderr, nowhere else.

That's why "If stdout is not a tty but stderr is, we write to stderr" sounds plain wrong. Maybe the intent is something different here?

Out of interest, what is `ts`? If that's something that multiplexes the command's stdout and stderr, how does that get demultiplexed on the other end?